### PR TITLE
[Silabs] Wi-fi: Cleanup wfx host events header file

### PR DESCRIPTION
**Pull request by bhmanda-silabs**
Duplicated `wfx_host_events.h` files removed and added common header for EFR and SiWx917

- Compiled all wifi boards - EFR mg24 (BRD4187C) +SiWX917 NCP , EFR mg24 (BRD4187C) + RS9116 , EFR mg24 (BRD4187C) + wf200 and BRD4338A SoC 
- Verified commissioning on all wifi boards (EFR mg24 (BRD4187C) + RS9116, EFR mg24 (BRD4187C) + wf200 , EFR mg24 (BRD4187C) + 917 NCP and 917 SoC BRD4338A)

| MACRO NAME | BOARD|
|--------|--------|
| RS911X_WIFI | RS9116, 917 NCP or 917 SoC|
| SLI_SI91X_MCU_INTERFACE | 917 SoC|
| EXP_BOARD| 917 NCP|
| WF200_WIFI | WF200|

### CODE REVIEW

File: examples/platform/silabs/SiWx917/BUILD.gn [hunk 1]
```diff
@@ -51,6 +51,7 @@ declare_args() {
 # Sanity check
 assert(chip_enable_wifi)
 silabs_plat_si91x_wifi_dir = "${chip_root}/src/platform/silabs/SiWx917/wifi"
+silabs_plat_dir = "${chip_root}/src/platform/silabs"
```
Comment (senthilku) [inline examples/platform/silabs/SiWx917/BUILD.gn:54]: move **silabs_plat_dir** above **silabs_plat_si91x_wifi_dir** and we can reuse this define?

File: src/platform/silabs/wifi/wfx_host_events.h [hunk 2]
```diff
@@ -351,40 +341,49 @@ void wfx_lwip_set_sta_link_up(void);
...
 #endif /* RS911X_WIFI */
 
 #ifdef WF200_WIFI
+bool wfx_is_sta_provisioned(void);
+sl_wfx_state_t wfx_get_wifi_state(void);
 void wfx_bus_start(void);
 sl_status_t get_all_counters(void);
 void sl_wfx_host_gpio_init(void);
 sl_status_t sl_wfx_host_process_event(sl_wfx_generic_message_t * event_payload);
 #endif
```
Comment (senthilku) [inline src/platform/silabs/wifi/wfx_host_events.h:379]: #endif /* #ifdef WF200_WIFI */

File: src/platform/silabs/wifi/wfx_host_events.h [hunk 3]
```diff
@@ -94,34 +95,52 @@ typedef struct __attribute__((__packed__)) sl_wfx_mib_req_s
...
 #if (SLI_SI91X_MCU_INTERFACE | EXP_BOARD)
 #include "sl_si91x_types.h"
 #include "sl_status.h"
 #include "sl_wifi_constants.h"
+#include "stdbool.h"
 
 #include "rsi_common_apis.h"
 #include "sl_wifi_device.h"
 
 #define SL_WIFI_ALLOCATE_COMMAND_BUFFER_WAIT_TIME_MS 1000
```
Comment (rosahay-silabs) [inline src/platform/silabs/wifi/wfx_host_events.h:111]:
```suggestion
#define SL_WIFI_ALLOCATE_COMMAND_BUFFER_WAIT_TIME_MS (1000)
```

File: src/platform/silabs/wifi/wfx_host_events.h [hunk 4]
```diff
@@ -131,101 +150,82 @@ typedef struct __attribute__((__packed__)) sl_wfx_mib_req_s
...
-
-// MAX SSID LENGTH excluding NULL character
-#define WFX_MAX_SSID_LENGTH (32)
-// MAX PASSKEY LENGTH including NULL character
-#define WFX_MAX_PASSKEY_LENGTH (64)
-
-#define CONVERT_SEC_TO_MS(TimeInS) (TimeInS * 1000)
-
 // WLAN MAX retry
 #define MAX_JOIN_RETRIES_COUNT 5
```
Comment (rosahay-silabs) [inline src/platform/silabs/wifi/wfx_host_events.h:154]:
```suggestion
#define MAX_JOIN_RETRIES_COUNT (5)
```

File: src/platform/silabs/wifi/wfx_host_events.h [hunk 5]
```diff
@@ -94,34 +95,52 @@ typedef struct __attribute__((__packed__)) sl_wfx_mib_req_s
     sl_wfx_mib_req_body_t body;
 } sl_wfx_mib_req_t;
 
-#else /* End WF200 else RS911x */
+#else /* End WF200 else 916,917 NCP and 917 SoC */
 
 #include "wfx_msgs.h"
 
 #if (SLI_SI91X_MCU_INTERFACE | EXP_BOARD)
```
Comment (rosahay-silabs) [inline src/platform/silabs/wifi/wfx_host_events.h:102]: isn't this condition redundant? considering
`SLI_SI91X_MCU_INTERFACE` for SiWx917 SoC,
`EXP_BOARD` is for rs911x and 917 NCP correct?

Comment (bhmanda-silabs) [reply src/platform/silabs/wifi/wfx_host_events.h:102]: `EXP_BOARD` is only for **917 NCP** and `SLI_SI91X_MCU_INTERFACE` for **SiWx917 SoC**

File: src/platform/silabs/wifi/wfx_host_events.h [hunk 6]
```diff
@@ -94,34 +95,52 @@ typedef struct __attribute__((__packed__)) sl_wfx_mib_req_s
...
-#else /* End WF200 else RS911x */
+#else /* End WF200 else RS9116,917 NCP and 917 SoC */
 
 #include "wfx_msgs.h"
 
 #if (SLI_SI91X_MCU_INTERFACE | EXP_BOARD)
 #include "sl_si91x_types.h"
 #include "sl_status.h"
 #include "sl_wifi_constants.h"
+#include "stdbool.h"
```
Comment (rosahay-silabs) [inline src/platform/silabs/wifi/wfx_host_events.h:106]:
```suggestion
#include <stdbool.h>
```
are we using some internal stdbool header ?

File: src/platform/silabs/wifi/wfx_host_events.h [hunk 7]
```diff
@@ -94,34 +95,52 @@ typedef struct __attribute__((__packed__)) sl_wfx_mib_req_s
...
+#define WFX_MAX_PASSKEY_LENGTH (SL_WIFI_MAX_PSK_LENGTH)
+#else
+// MAX SSID LENGTH excluding NULL character
+#define WFX_MAX_SSID_LENGTH (32)
+// MAX PASSKEY LENGTH including NULL character
+#define WFX_MAX_PASSKEY_LENGTH (64)
+#endif // (SLI_SI91X_MCU_INTERFACE  | EXP_BOARD)
+
+#include "sl_status.h"
+#include "stdbool.h"
```
Comment (rosahay-silabs) [inline src/platform/silabs/wifi/wfx_host_events.h:134]:
```suggestion
#include <stdbool.h>
```

File: src/platform/silabs/wifi/wfx_host_events.h [hunk 8]
```diff
@@ -94,34 +95,52 @@ typedef struct __attribute__((__packed__)) sl_wfx_mib_req_s
...
-#define SL_WFX_DISCONNECT_IND_ID 3
-#define SL_WFX_SCAN_COMPLETE_ID 4
-#endif /* WF200 */
+#define SL_WFX_STARTUP_IND_ID (1)
+#define SL_WFX_CONNECT_IND_ID (2)
+#define SL_WFX_DISCONNECT_IND_ID (3)
+#define SL_WFX_SCAN_COMPLETE_ID (4)
+#endif /* WF200_WIFI */
+
+#if (SLI_SI91X_MCU_INTERFACE | EXP_BOARD)
```
Comment (rosahay-silabs) [inline src/platform/silabs/wifi/wfx_host_events.h:121]: the block below can be included in the above `#else` part if I am not wrong.

**Review by rosahay-silabs — APPROVED** [review]

**Review by senthilku — APPROVED** [review]

File: src/platform/silabs/wifi/wfx_host_events.h (hunk 6 above)
Comment (bhmanda-silabs) [reply src/platform/silabs/wifi/wfx_host_events.h]: done @rosahay-silabs

File: src/platform/silabs/wifi/wfx_host_events.h (hunk 7 above)
Comment (bhmanda-silabs) [reply src/platform/silabs/wifi/wfx_host_events.h]: done

File: src/platform/silabs/wifi/wfx_host_events.h (hunk 8 above)
Comment (bhmanda-silabs) [reply src/platform/silabs/wifi/wfx_host_events.h]: Refactored in a cleaner way for 917 SoC and NCP

**Review by jmartinez-silabs — COMMENTED** [review]
This looks ok, but it most a bunch of conditional includes and defines and what not. 
You pr description doesn't clearly state what was built and tested.

Did you build different apps and different configurations? did you test them at run time?

**Review by jmartinez-silabs — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Pull request merged**